e2e/pageobjects/ide/Editor.ts

```diff
@@ -179,10 +230,126 @@ export class Editor {
+        const errorInLineLocator: By = await this.getErrorInLineLocator(lineNumber);
+
+        await this.driverHelper.waitDisappearanceWithTimeout(errorInLineLocator, timeout);
+    }
+
+    async waitStoppedDebugBreakpoint(tabTitle: string, lineNumber: number, timeout: number = TestConstants.TS_SELENIUM_DEFAULT_TIMEOUT) {
+        const stoppedDebugBreakpointLocator: By = By.xpath(await this.getStoppedDebugBreakpointXpathLocator(tabTitle, lineNumber));
+
+        await this.driverHelper.waitVisibility(stoppedDebugBreakpointLocator, timeout);
+    }
+
+    async waitBreakpoint(tabTitle: string, lineNumber: number, timeout: number = TestConstants.TS_SELENIUM_DEFAULT_TIMEOUT) {
+        const debugBreakpointLocator: By = await this.getDebugBreakpointLocator(tabTitle, lineNumber);
+
+        await this.driverHelper.waitVisibility(debugBreakpointLocator, timeout);
+    }
+
+    async waitBreakpointAbsence(tabTitle: string, lineNumber: number, timeout: number = TestConstants.TS_SELENIUM_DEFAULT_TIMEOUT) {
+        const debugBreakpointLocator: By = await this.getDebugBreakpointLocator(tabTitle, lineNumber);
+
+        await this.driverHelper.waitDisappearanceWithTimeout(debugBreakpointLocator, timeout);
+    }
+
+    async waitBreakpointHint(tabTitle: string, lineNumber: number, timeout: number = TestConstants.TS_SELENIUM_DEFAULT_TIMEOUT) {
+        const debugBreakpointHintLocator: By = await this.getDebugBreakpointHintLocator(tabTitle, lineNumber);
+
+        await this.driverHelper.waitVisibility(debugBreakpointHintLocator, timeout);
+    }
+
+    async waitBreakpointHintDisappearance(tabTitle: string, lineNumber: number, timeout: number = TestConstants.TS_SELENIUM_DEFAULT_TIMEOUT) {
+        const debugBreakpointHintLocator: By = await this.getDebugBreakpointHintLocator(tabTitle, lineNumber);
+
+        await this.driverHelper.waitDisappearanceWithTimeout(debugBreakpointHintLocator, timeout);
+    }
+
+    async activateBreakpoint(tabTitle: string,
+        lineNumber: number,
+        timeout: number = TestConstants.TS_SELENIUM_DEFAULT_TIMEOUT) {
+
+        const attempts: number = TestConstants.TS_SELENIUM_DEFAULT_ATTEMPTS;
+        const polling: number = TestConstants.TS_SELENIUM_DEFAULT_POLLING;
+
+        for (let i = 0; i < attempts; i++) {
+            const elementLocator: By = await this.getLineNumberBlockLocator(tabTitle, lineNumber);
+            const element: WebElement = await this.driverHelper.waitVisibility(elementLocator, timeout);
+
+            try {
+                await this.driverHelper.getAction().mouseMove(element, { x: 5, y: 5 }).perform();
+                await this.waitBreakpointHint(tabTitle, lineNumber);
+                await this.driverHelper.getAction().click().perform();
+                await this.waitBreakpoint(tabTitle, lineNumber);
+                return;
+            } catch (err) {
+                // ignore errors and wait
+                await this.driverHelper.wait(polling);
```

Comment:
Do we know what errors can happen here? Shouldn't we at least log them?

Reply:
In this case, we should ignore all exceptions

Reply:
Because actions inside of try/catch are very sensitive, and I tried to process most common exceptions and final was that the "WebDriverError" may be happening and it does not make sense to process other exceptions because it is a parent of all driver exceptions. Anyway, if something went wrong exception with full explanation will be thrown in the further code.